File: my_py_pkg/my_py_pkg/my_first_publisher.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class MyFirstPublisher(Node):

    def __init__(self):
        super().__init__("py_publisher")
        self.counter_ = 0
        self.publisher = self.create_publisher(String, "string_messages", 10)
        self.get_logger().info("Hello ROS2")
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        self.counter_ += 1
        data = String()
        data.data = "Hello " + str(self.counter_)
        self.publisher.publish(data)
        self.get_logger().info("Hello " + str(self.counter_))


def main(args=None):
    rclpy.init(args=args)
    node = MyFirstPublisher()
    
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class MyFirstPublisher : public rclcpp::Node
{
public:
    MyFirstPublisher() : Node("cpp_publisher"), counter_(0)
    {
        RCLCPP_INFO(this->get_logger(), "Hello from %s", this->get_name());

        publisher = this->create_publisher<example_interfaces::msg::String>("string_messages", 10);

        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                         std::bind(&MyFirstPublisher::timerCallback, this));
    }

private:
    
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
    void timerCallback()
    {
        counter_++;
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
        auto data = example_interfaces::msg::String();
        data.data = std::string("Hello ") + std::to_string(counter_);
        publisher->publish(data);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyFirstPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
